impl/command_line/options.cxx

```diff
@@ -184,28 +184,58 @@ namespace substrate::commandLine
 		}, _option);
 	}
 
+	[[nodiscard]] inline std::string_view option_t::typeToValue() const noexcept
+	{
+		switch (_valueType){
+			case optionValueType_t::signedInt:
+				return "INT"sv;
+			case optionValueType_t::unsignedInt:
+				return "UINT"sv;
+			case optionValueType_t::boolean:
+				return "BOOL"sv;
+			case optionValueType_t::string:
+				return "STRING"sv;
+			case optionValueType_t::path:
+				return "PATH"sv;
+			case optionValueType_t::userDefined:
+				return "VAL"sv;
+		}
+	}
+
 	[[nodiscard]] std::string option_t::displayName() const noexcept
 	{
 		if (_option.valueless_by_exception())
 			return ""s;
+		const auto typeValue = [&]() -> std::string {
+			if (!takesParameter())
+				return {};
+			return " "s + std::string{typeToValue()};
+		}();
 		return std::visit(match_t
 		{
-			[](const std::string_view &option) { return std::string{option}; },
-			[](const optionFlagPair_t &option)
-				{ return std::string{option._shortFlag} + ", "s + std::string{option._longFlag}; },
-			[](const optionValue_t &option) { return std::string{option.metaName()}; },
+			[&typeValue](const std::string_view &option) { return std::string{option} + typeValue ; },
```

Comment:
Saying only `[&]` in the capture should still only capture the one used value - also, there's a stray space just before the `;`.

Same comment for the net capture. The `this` capture is fine though - that we understand precisely and is fine.